### ISSUE TITLE
Starting transport with two synced Players starts them at the wrong time.

### ISSUE THREAD

**Issue by cristiano-belloni** [Tonejs/Tone.js#540]
This might just be not understanding how `Transport`s work, but I read the manual and the wiki (and a couple of bug reports here) and there's nothing that makes me think I'm doing it wrong. If I am, sorry :)

I have a Codesandbox to repro the code [here](https://codesandbox.io/s/blazing-pine-72zfw).

What I'm trying to do is: having 2 Players, I want to schedule them (in TransportTime) within the Transport, then start the Transport at some point in time. I expect that, if I `sync()` the two Players, `start()` them with a startTime argument and finally start the Transport, they should be playing at their respectively correct startTime after I started the Transport. In other words, if I do:

```javascript
    player1
      .toMaster()
      .sync()
      .start(1.555)

    player2
      .toMaster()
      .sync()
      .start(4.1)

    Tone.Transport.start("+0")
```

I expect player1 to play `1.555` seconds after the Transport is started, and player2 to play `4.1` seconds after the transport is started. For simplicity's sake, in the example I start the `Transport` immediately (in Context time), but if I started it in `+1` seconds, I'd expect to hear player1 `2.555` seconds later and player2 `5.1` seconds later.

What I hear, instead, are **the two players starting at the same time, after some 12 - 13 seconds** (again, see the example project [here](https://codesandbox.io/s/blazing-pine-72zfw)).

Is that a bug or am I getting it completely wrong?

**Comment by tambien** [comment]
@cristiano-belloni did you find what the issue was?

**Comment by cristiano-belloni** [comment]
@tambien - I closed because it's incredibly confusing and I don't know how to repro / express this well. I have now [this sandbox](https://codesandbox.io/s/little-voice-9qu03), where I use `Tone.Time` and `Tone.TransportTime` everywhere, and it *seems* to work, except the Player starting at `0` actually starts a bit later. This is kinda making me crazy.

Could you please just run both of them, just to say if it's the result you would expect??

**Comment by tambien** [comment]
Your code looks fine to me. This is kinda weird, but i think the issue is somehow with codesandbox. [With this example](https://codesandbox.io/s/blazing-pine-72zfw), I notice a difference in behavior if i do a full page refresh and click the start button, it seems to work fine the first time after the page is loaded, but if i just click the in-page refresh button, above the browser window, i get the incorrect behavior that you mentioned. Let me know if you are hearing the same thing. 

Also, I tested this same code in a different environment and it also worked as expected. This kind of bug will drive a person crazy. I wonder what could cause this error. something maybe to do with the way they loaded the wav files? 

If you can verify the behavior that i mentioned where it works on full page refresh but not after an in-page refresh, then we should file a bug with codesandbox.

**Comment by cristiano-belloni** [comment]
@tambien - I don't think it's codesandbox.

I get the same behaviour at home, and I condensed my home example in the sandboxes you see. i think you're getting different results because the hot reload is somehow broken in codesandbox, but if you 1) open the project in a different page (there's an "Open in new Window" button on the top right and 2) refresh every time, you will notice the strange behaviours.

Or you can try to open it in your personal PC, I think it will kinda do the same.

**Comment by tambien** [comment]
[Here's a fork of your example](https://codesandbox.io/s/misty-architecture-bw7zx). I added a call to `Tone.start()` and wait for the promise to resolve before loading and playing the samples. This seem to be getting the correct behavior.

**Comment by cristiano-belloni** [comment]
@tambien - your fork doesn't work for me. It plays the first sound after a while, and the second sound immediately after.

**Comment by cristiano-belloni** [comment]
@tambien - I think I know why it sounds good for you - because you click the button immediately after you refresh the page. If you wait several seconds, then click it, it will sound strange. It's as if it's picking up the Context time and not the Transport time, but in a weird way. See this video that I just recorded with your fork (zip because Github won't let me upload .mov files)

[bug-tone.mov.zip](https://github.com/Tonejs/Tone.js/files/3581297/bug-tone.mov.zip)

**Comment by tambien** [comment]
You're absolutely right. thanks for bearing with me on this. I think i found the cause of the error, seems like i introduced an error at some point to `Tone/source/Source.js`, i'm trying to track that down now.

**Comment by tambien** [comment]
[This is the line which caused the error](https://github.com/Tonejs/Tone.js/commit/0ba72ac93acf0da45119129df686cee724c52b9b#diff-716790d480ea74314edbbbb88e4aafbeR149). You were right that it was clamping to the currentTime in this scenario.

**Comment by cristiano-belloni** [comment]
Wow, happy that you found the issue - these timing bugs are so tricky.

**Comment by tambien** [comment]
ok feels like it works now with `tone@13.8.27`. If you have a second could you verify the fix and close the issue if it's working for you. 

i'll publish this version as latest if it all works.

**Comment by cristiano-belloni** [comment]
Late here and I'm afk already, but will try first thing in the morning and confirm. Thanks!

**Comment by cristiano-belloni** [comment]
@tambien - timing seems solid now. Many thanks again!

**Comment by SalsaAddict** [comment]
Hi, this thread says that the bug was fixed in tone@13.8.27 but the latest version is tone@13.8.25 and next is 14....

I urgently need a fix for this bug. Where can I download 13.8.27?

**Comment by tambien** [comment]
@SalsaAddict try `npm install tone@13.8.27` or if you're looking for just the build file, you can find it on [unpkg](https://unpkg.com/tone@13.8.27/build/Tone.js)
